### ISSUE TITLE
Executor container exits with status 2 when exceeding the memory limit

### ISSUE THREAD

**Issue by Fitzse** [argoproj/argo#724]
BUG REPORT

**What happened**:
When trying to upload a large folder (7223 files and ~1GB of data) as an output artifact of a step, the `wait` container exits with status code 2. This manifests as no error in the wait container logs and the `message: failed to save outputs: verify serviceaccount default:default has necessary privileges` error in the workflow-controller logs.

**What you expected to happen**:
The output artifact should be uploaded successfully to S3.

**How to reproduce it (as minimally and precisely as possible)**:
Will try and generate a test case that does not involve our private git repository.

**Anything else we need to know?**:
After some debugging with @jessesuen we found that removing the resource limits below would allow the operation to complete successfully. https://github.com/argoproj/argo/blob/98a44c99c2515f2295327ae9572732586ddc3d7b/workflow/controller/workflowpod.go#L249-L256

The max memory limit of 512mb was too small for this operation (in the successful test without a limit it took ~582mb peak). This memory spike happens during the `docker cp` and `gzip` commands in `archivePath`:
https://github.com/argoproj/argo/blob/fdafbe27e5e2f4f2d58913328ae22db9a6c363b4/workflow/executor/executor.go#L432-L441

**Environment**:
- Argo version:
```
argo: v2.0.0
  BuildDate: 2018-02-06T21:38:42Z
  GitCommit: 0978b9c61cb7435d31ef8d252b80e03708a70adc
  GitTreeState: clean
  GitTag: v2.0.0
  GoVersion: go1.9.1
  Compiler: gc
  Platform: darwin/amd64
```
- Kubernetes version :
```
clientVersion:
  buildDate: 2018-01-18T21:11:08Z
  compiler: gc
  gitCommit: 5fa2db2bd46ac79e5e00a4e6ed24191080aa463b
  gitTreeState: clean
  gitVersion: v1.9.2
  goVersion: go1.9.2
  major: "1"
  minor: "9"
  platform: darwin/amd64
serverVersion:
  buildDate: 2017-11-20T05:17:43Z
  compiler: gc
  gitCommit: 9befc2b8928a9426501d3bf62f72849d5cbcd5a3
  gitTreeState: clean
  gitVersion: v1.8.4
  goVersion: go1.8.3
  major: "1"
  minor: "8"
  platform: linux/amd64
```

**Other debugging information (if applicable)**:
- executor logs:
```
time=“2018-02-07T00:24:47Z” level=info msg=“Waiting on main container”
time=“2018-02-07T00:24:48Z” level=info msg=“main container started with container ID: 1cd03a5101ccab795b8094912a0b761356768fde2cf97f67be58bba406070f3d”
time=“2018-02-07T00:24:48Z” level=info msg=“Starting annotations monitor”
time=“2018-02-07T00:24:48Z” level=info msg=“docker wait 1cd03a5101ccab795b8094912a0b761356768fde2cf97f67be58bba406070f3d”
time=“2018-02-07T00:24:48Z” level=info msg=“Starting deadline monitor”
time=“2018-02-07T00:28:55Z” level=info msg=“Main container completed”
time=“2018-02-07T00:28:55Z” level=info msg=“No sidecars”
time=“2018-02-07T00:28:55Z” level=info msg=“Saving output artifacts”
time=“2018-02-07T00:28:55Z” level=info msg=“Saving artifact: repository”
time=“2018-02-07T00:28:55Z” level=info msg=“Archiving 1cd03a5101ccab795b8094912a0b761356768fde2cf97f67be58bba406070f3d:/src/SLL to /argo/outputs/artifacts/repository.tgz”
time=“2018-02-07T00:28:55Z” level=info msg=“Annotations monitor stopped”
time=“2018-02-07T00:28:55Z” level=info msg=“sh -c docker cp -a 1cd03a5101ccab795b8094912a0b761356768fde2cf97f67be58bba406070f3d:/src/SLL - | gzip > /argo/outputs/artifacts/repository.tgz”
time=“2018-02-07T00:28:56Z” level=info msg=“Deadline monitor stopped”
time=“2018-02-07T00:29:47Z” level=info msg=“Alloc=2482 TotalAlloc=27503 Sys=9830 NumGC=13 Goroutines=8"
```

**Comment by jessesuen** [comment]
Fixed in master and release-2.0 branches

**Comment by Fitzse** [comment]
@jessesuen Thanks for the quick turn-around on this! I'll build a controller image of the `release-2.0` branch and start using it.
